File: leetcode/medium/1441. Build an Array With Stack Operations/main.go
package main

const (
	PUSH = "Push"
	POP  = "Pop"
)

func buildArray(target []int, n int) []string {
	result := []string{}

	curr := 1
	idx := 0

	for curr <= n && idx < len(target) {
		num := target[idx]
		if num == curr {
			result = append(result, PUSH)
			curr++
			idx++
		} else {
			result = append(result, PUSH)
			result = append(result, POP)
			curr++
		}

	}

	return result
}

File: leetcode/medium/1441. Build an Array With Stack Operations/main.rs
impl Solution {
    pub fn build_array(target: Vec<i32>, n: i32) -> Vec<String> {
        let mut result = Vec::new();

        let mut curr_number = 1;
        let mut idx = 0;

        while curr_number <= n && idx < target.len() {
            let number = target[idx];

            if number == curr_number {
                result.push(String::from("Push"));
                idx += 1;
            } else {
                result.push(String::from("Push"));
                result.push(String::from("Pop"));
            }

            curr_number += 1;
        }

        result
    }
}
